### DISCUSSION TITLE
Choosing a simulator: what can be simulated at Carla

### DISCUSSION THREAD

**Discussion by laercio-barbosa** [carla-simulator/carla#6170] [Q&A]
Hello all. 

I am designing a lateral controller for Autonomous Trucks and looking for a simulator that can performs/simulate the following influences in the vehicle dynamics:

1. Changes in road behavior (friction coefficient) in on-road lanes under differente weather conditions (sunny day, rain, snow)
    A truck can be submited to different error mainly when is performing turn manouver under these conditions.

2. Changes in road behavior (friction coefficient) in off-road terrain under differente weather conditions (sunny day, rain, snow)
    A truck can be submited to different error mainly when is performing turn manouver under these conditions.

3. Road imperfections (potholes, puddles, banking)
    The steering components and controllers of a truck will face disturbances when submited to these conditions taking the vehicle to a different trajectory.

4. Wind effects such as lateral forces and/or drag resistance in the vehicle
    The steering components and controllers of a truck will face disturbances when submited to these conditions taking the vehicle to a different trajectory.

5. Vehicle parts degradation/ageing (e.g.: a flat or worn out tire, steering wheel backlash or electric power steering errors) 
    The steering components and controllers of a truck will face disturbances when submited to these conditions taking the vehicle to a different trajectory.

How does this simulator manage these points?
Is there any way to “fake” one or all of them (like decrease tire diameter to fake a flat tire, noise in sterring signal to simulate steering wheel backlash, change Truck position to simulate a skidding)?

Thank you all

**Comment by enter802** [comment]
Did you figure out how to simulate any/all of the above behaviours? I am looking to do the same. Let me know. Thanks

**Reply by laercio-barbosa** [reply]
No, unfortunately not. Then I moved to DYNA4 from Vector as my master's advisor had a money from a project to invest in a academic license (+- 200 Euros).

**Reply by enter802** [reply]
I see. I have tried changing weather conditions, but it seems like there is no affect on the vehicle dynamics in Carla. Finding it hard to test my controller's robustness due to uncertainties. Thanks anyway and best of luck!

**Reply by enter802** [reply]
How about these options [here](https://carla.readthedocs.io/en/latest/tuto_G_control_vehicle_physics/)? For example changing drag coefficient, tire friction through vehicle physics. What are your thoughts? They seem to be pretty good alternatives to test robustness of controllers. I believe the weather conditions are geared towards perception robustness.

**Reply by laercio-barbosa** [reply]
- For a wind of 30m/s crossing the road in 45 degrees, what vehicle parameters you need to change to fake the wind force impact?
- What a banking angle brings to your controller when the vehicle steering changes due a composition of gravity and circular motion dynamics?
- Also, what kind of tire model Carla use? I mean, a tire suffers deformation in curves and in high speeds and this changes the friction coefficient. You see, it is not too simple.

**Reply by enter802** [reply]
I understand you are trying to integrate quite a variety of complex uncertainties into your model whereas I am just looking for any kind of uncertainty/change for vehicle environment. I am glad you found your way! Best of luck
